### ISSUE TITLE
Zos says no disks but seems to have created file systems

### ISSUE THREAD

**Issue by scottyeager** [threefoldtech/zos#2339]
One farmer reported that their node was showing logs like this on first boot:

![image](https://github.com/threefoldtech/zos/assets/17682591/3ea44832-3192-4a2e-9a85-461b9d7fe157)

I suspected that maybe the disks had not been properly wiped, so asked the farmer to try again. They showed the result of this attempt:

![image](https://github.com/threefoldtech/zos/assets/17682591/befa7860-479e-49a5-9b64-ae563f8b56cd)

But node continues showing no disks:

![image](https://github.com/threefoldtech/zos/assets/17682591/14858426-c0ad-43fd-a89a-e2b6fed20987)

Two things struck me:

1. Seems farmer followed the correct procedure with `wipefs`
2. On the second wipe, the disks are partitionless `btrfs`, so it really looks like Zos put file systems on these disks and then refused to use them for some reason

**Comment by shng4you** [comment]
Here is another farmer report with the same issue: After wiping with grml, the error remain the same.
![afbeelding_1029hxv](https://github.com/threefoldtech/zos/assets/95933607/b9e4294e-a37f-4929-8d95-d9bbd12757e4)

**Comment by scottyeager** [comment]
We are hearing quite a few more reports of this.

**Comment by muhamadazmy** [comment]
Okay, I will need some help from the farmer to debug this since there is no way we can do this remotely. 
This requires a farmer with some IT experience since we will need him to have:

## requirements
- github account
- his ssh key configured on his github account. He will need to ssh to the node later.

# Steps
## Booting the node
- Build a new zos image but for devnet. He needs then to make sure `ssh-user` is passed as a kernel param like in screenshot
![image](https://github.com/threefoldtech/zos/assets/10920323/a87721b5-3c95-4a6d-9d19-757a88f1e426)
- Any valid farm id on devnet is good, if you don't have your own you can use the one in the screenshot
- Use the image format you like. I think USB image is the easiest
- Boot the node on devnet 
- During boot (early stages) you can see what IP the node got in the logs. Take a note of that (might help to record a video in case it rolls away too quickly)
- When the node starts to give error it's already ready to accept your ssh

## Debugging
now all you need to do is to ssh to the node and inspect the issue as follows:
```sh
ssh root@<node-ip>
```
Once inside, some things you can do is to run
```
lsblk
```
and take a node (a photo) of the output. This will give us enough information to see what the zos kernel sees. If this does not show any disks (or fewer disks than it should) we might also use the output from `dmesg` 

If output from dmesg is too much (and I assume it is) you can save that to a file, and then copy it to your laptop (with scp) then include this in the issue as well.

```sh
dmesg > /tmp/dmesg.out
```
then on your laptop do
```sh
scp root@<node-ip>:/tmp/dmesg.out .
```
this will bring the file over.

**Comment by scottyeager** [comment]
Hi @muhamadazmy,

I've done as you directed on one of my own nodes. I wiped the disks and booted it to devnet, and now there are no disks detected.

![image](https://github.com/threefoldtech/zos/assets/17682591/65d35e73-16eb-4e96-b0c4-07d883a00066)

Turns out the SSD is in fact mounted by Zos:

![image](https://github.com/threefoldtech/zos/assets/17682591/34382e6f-d526-4925-9b84-03afb2071a56)

It's empty:

![image](https://github.com/threefoldtech/zos/assets/17682591/ac273bf1-11e6-4c3e-8b33-bb2a2d7b134b)

**Comment by Mik-TF** [comment]
I followed the process without wiping the disk. Here's the output:

![image](https://github.com/threefoldtech/zos/assets/77026219/323717ab-03a5-4fa2-8e83-5acc0ac800b5)

And I get this error from the zos terminal:

![image](https://github.com/threefoldtech/zos/assets/77026219/bf1d0a17-87a8-4c69-9adb-3a7a0251495d)

**Comment by muhamadazmy** [comment]
That's weird! Is there a way i can access this nose? Can you configure a port forward or something so i can hop in!

Can you also do an 'ls -a' on that mounted disk and see if there is a hidden file, and what the contents are

**Comment by scottyeager** [comment]
> Is there a way i can access this nose?

Coordinated via DM.

**Comment by muhamadazmy** [comment]
Okay after managing to have access to the affected node i was able to run an investigation and here is what I found:
- Seems that btrfs behaviour has some small changes on how it work with quota and quota groups
- Code of zos was not aware of this slight change, it caused the node to report random storage error. For some reason my local development node was not affected! 
- Because this change is in base image, this will also affect mainnet. Hence I will need to patch mainnet after making sure this fix is working find on devnet
- I just pushed a fix to devnet. I manually rebooted your node, I would love to get feedback on the status of the node since i can not access it anymore (again)

Once we sure this fix is working find on devnet, I will patch mainnet

**Comment by muhamadazmy** [comment]
@ashraffouda will be available Sunday. Please check the status of this issue before updating mainnet. I already cherry picked the fix to v3.10.x branch. If the issue is fixed make sure to create a new release from v3.10.x (it should be v3.10.5) before creating the motion to update mainnet
